### PR TITLE
Packet provider

### PR DESCRIPTION
**Pull request by rfranzke**
**What this PR does / why we need it**:
Adds infrastructure and control plane controller for Packet provider.

Also, it simplifies some parts of the control plane actuators of AWS and GCP.

**Special notes for your reviewer**:
Based on https://github.com/gardener/gardener/pull/916.
The control plane controller implements the CCM deployment. Based on how https://github.com/gardener/gardener/pull/916 will continue I will add CSI deployment and possibly webhooks later.

**Release note**:

```noteworthy operator
Added Packet infrastructure controller
```

### CODE REVIEW

File: controllers/provider-aws/charts/internal/cloud-controller-manager/templates/cloud-controller-manager.yaml [hunk 1]
```diff
@@ -60,7 +60,7 @@ spec:
         - --cluster-cidr={{ .Values.podNetwork }}
         - --cluster-name={{ .Values.clusterName }}
         - --concurrent-service-syncs=10
-        - --configure-cloud-routes={{ .Values.configureRoutes }}
+        - --configure-cloud-routes=false
```
Comment (adracus) [inline controllers/provider-aws/charts/internal/cloud-controller-manager/templates/cloud-controller-manager.yaml]: Leftover / why is this changed in the Packet PR?

File: controllers/provider-packet/pkg/controller/infrastructure/actuator_reconcile.go [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+		return &controllererrors.RequeueAfterError{
+			Cause:        err,
+			RequeueAfter: 30 * time.Second,
+		}
+	}
+
+	return a.updateProviderStatus(ctx, tf, infrastructure)
+}
+
+func generateTerraformInfraConfig(infrastructure *extensionsv1alpha1.Infrastructure, projectID string) map[string]interface{} {
```
Comment (adracus) [inline controllers/provider-packet/pkg/controller/infrastructure/actuator_reconcile.go]: Make this public and add tests

File: controllers/provider-packet/pkg/packet/client/client.go [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+	"github.com/packethost/packngo"
+)
+
+type packetClient struct {
+	packet *packngo.Client
+}
+
+// NewClient creates a new Client for the given Packet credentials
+func NewClient(packetAPIKey string) ClientInterface {
```
Comment (adracus) [inline controllers/provider-packet/pkg/packet/client/client.go]: Is this still WIP?

File: controllers/provider-aws/charts/internal/cloud-controller-manager/templates/cloud-controller-manager.yaml (hunk 1 above)
Comment (rfranzke) [reply controllers/provider-aws/charts/internal/cloud-controller-manager/templates/cloud-controller-manager.yaml]: Created https://github.com/gardener/gardener-extensions/pull/80

File: controllers/provider-packet/pkg/packet/client/client.go (hunk 3 above)
Comment (rfranzke) [reply controllers/provider-packet/pkg/packet/client/client.go]: No, see gardener/gardener#916, especially https://github.com/gardener/gardener/pull/916#discussion_r281515425. According to this, @deitch wants to keep the client in case he wants to use it in the future.

As he will take over ownership of the packet provider I just added it here as well.

**Comment by rfranzke** [comment]
Any more feedback?
